frontend/src/scenes/pipeline/BatchExportBackfillModal.tsx

```diff
@@ -62,43 +71,74 @@ export function BatchExportBackfillModal({ id }: BatchExportRunsLogicProps): JSX
+                                }
+                                onChange={(date) => {
+                                    if (date) {
+                                        let projectDate = date.tz(timezone, true)
 
-                                    if (batchExportConfig && batchExportConfig.interval === 'day') {
-                                        projectDate = projectDate.hour(0).minute(0).second(0)
-                                    }
+                                        if (batchExportConfig && batchExportConfig.interval === 'day') {
+                                            projectDate = projectDate.hour(0).minute(0).second(0)
+                                        }
 
-                                    onChange(projectDate)
-                                } else {
-                                    onChange(date)
-                                }
-                            }}
-                            placeholder="Select start date"
-                            granularity={
-                                batchExportConfig
-                                    ? batchExportConfig.interval === 'hour'
-                                        ? 'hour'
-                                        : batchExportConfig.interval.endsWith('minutes')
-                                        ? 'minute'
+                                        onChange(projectDate)
+                                    } else {
+                                        onChange(date)
+                                    }
+                                }}
+                                placeholder="Select start date"
+                                granularity={
+                                    batchExportConfig
+                                        ? batchExportConfig.interval === 'hour'
+                                            ? 'hour'
+                                            : batchExportConfig.interval.endsWith('minutes')
+                                            ? 'minute'
+                                            : 'day'
                                         : 'day'
-                                    : 'day'
-                            }
-                        />
-                    )}
+                                }
+                            />
+                        ) : (
+                            <LemonInput value="Earliest available" disabled />
+                        )
+                    }
                 </LemonField>
+
+                {batchExportConfig?.model == 'persons' ? (
+                    <LemonField name="earliest_backfill">
+                        {({ onChange }) => (
+                            <LemonCheckbox
+                                bordered
+                                label={
+                                    <span className="flex items-center gap-2">
+                                        Backfill since earliest available
```

Comment:
I wonder whether _"Backfill all {model}"_ would be better suited here? If not that, then _"backfill since the beginning of time"_ is certainly better - the current language is a bit hard to understand

Reply:
I agree, I think it would also be consistent to use "beginning of time" everywhere. If you don't mind though, I'll merge this PR since CI is taking a while and I care about the backend changes making their way in. I'll open a follow-up PR to update the naming here as soon as this is merged.